src/main/java/ua/com/juja/microservices/keepers/slackbot/model/SlackParsedCommand.java

```diff
@@ -7,89 +7,76 @@
-        usersCopy.putAll(users);
-        usersCopy.remove(fromSlackName);
-        logger.debug("Found users {} in text: [{}]", users.toString(), text);
-        return new LinkedList(usersCopy.values());
+        return usersInText;
     }
 
     public UserDTO getFirstUserFromText() {
-        checkSlackNamesPresence();
-        if(users.isEmpty()){
-            logger.warn("The map 'users' is empty");
-            throw new WrongCommandFormatException("The map 'users' is empty");
+        if (usersInText.size() == 0) {
+            logger.warn("The text: '{}' doesn't contain any slack names", text);
+            throw new WrongCommandFormatException(String.format("The text '%s' doesn't contain any slack names", text));
+        } else {
+            return usersInText.get(0);
         }
-        UserDTO firstUser = users.get(slackNamesInText.get(0));
-        logger.debug("Found user: {} in text: [{}]", firstUser.toString(), text);
-        return firstUser;
     }
 
-    public String getTextWithoutSlackNames(){
+    public String getTextWithoutSlackNames() {
         String result = text.replaceAll(SLACK_NAME_PATTERN, "");
-        result = result.replaceAll("\\s+"," ").trim();
+        result = result.replaceAll("\\s+", " ").trim();
         return result;
     }
 
     public UserDTO getFromUser() {
-        return users.get(fromSlackName);
+        return fromUser;
     }
 
     public String getText() {
         return text;
     }
 
     public int getUserCountInText() {
-        return userCount;
+        return usersInText.size();
     }
 
-    private List<String> getSlackNames() {
-        List<String> result = new ArrayList<>();
-        Pattern pattern = Pattern.compile(SLACK_NAME_PATTERN);
-        Matcher matcher = pattern.matcher(text);
-        while (matcher.find()) {
-            result.add(matcher.group());
-        }
-        return result;
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        SlackParsedCommand that = (SlackParsedCommand) o;
+
+        if (fromUser != null ? !fromUser.equals(that.fromUser) : that.fromUser != null) return false;
+        if (text != null ? !text.equals(that.text) : that.text != null) return false;
+        return usersInText != null ? usersInText.equals(that.usersInText) : that.usersInText == null;
     }
 
-    private void checkSlackNamesPresence() {
-        if (userCount == 0) {
-            logger.warn("The text: [{}] doesn't contain slack name", text);
-            throw new WrongCommandFormatException(String.format("The text '%s' doesn't contain any slackName", text));
-        }
+    @Override
+    public int hashCode() {
+        int result = fromUser != null ? fromUser.hashCode() : 0;
+        result = 31 * result + (text != null ? text.hashCode() : 0);
+        result = 31 * result + (usersInText != null ? usersInText.hashCode() : 0);
+        return result;
```

Comment:
You can just use  anotation  @EqualsAndHashCode of lombok.
And add the Annotation to UserDTO